File: web_server.py
# https://github.com/viktorexe/terristats-discord-bot
from aiohttp import web
import asyncio
async def health_check(request):
    return web.Response(text="Bot is running", status=200)
async def start_web_server(port=8000):
    app = web.Application()
    app.router.add_get('/', health_check)
    app.router.add_get('/health', health_check)
    
    runner = web.AppRunner(app)
    await runner.setup()
    site = web.TCPSite(runner, '0.0.0.0', port)
    await site.start()
    print(f"Web server started on port {port}")

File: webServer.js
// https://github.com/viktorexe/terristats-discord-bot
// Tiny HTTP server used purely for health checks. Hosting providers such as
// Railway ping "/" (or "/health") to confirm the process is still alive.
const http = require('http');

/**
 * Start a minimal health-check web server.
 * @param {number} port - Port to listen on (defaults to 8000).
 * @returns {Promise<http.Server>} the running server instance.
 */
function startWebServer(port = 8000) {
  return new Promise((resolve) => {
    const server = http.createServer((req, res) => {
      if (req.url === '/' || req.url === '/health') {
        res.writeHead(200, { 'Content-Type': 'text/plain' });
        res.end('Bot is running');
      } else {
        res.writeHead(404, { 'Content-Type': 'text/plain' });
        res.end('Not found');
      }
    });

    server.listen(port, '0.0.0.0', () => {
      console.log(`Web server started on port ${port}`);
      resolve(server);
    });
  });
}

module.exports = { startWebServer };
